### PR TITLE
Add push_server.disable_tls_validation to Dendrite configuration

### PR DESCRIPTION
**Pull request by tommie**
This disables the TLS certificate check for Push gateways. Sytest uses HTTPS, but with a self-signed certificate. Synapse already has an option like this.

Dendrite-side must be committed first: https://github.com/matrix-org/dendrite/pull/2061

### CODE REVIEW

**Comment by tommie** [comment]
Reviewer: @kegsay

File: lib/SyTest/Homeserver/Dendrite.pm [hunk 1]
```diff
@@ -101,6 +101,7 @@ sub _get_config
          server_name => $self->server_name,
          private_key => $self->{paths}{matrix_key},
          presence_enabled => $JSON::true,
+         http_client_tls_insecure_skip_verify_for_testing => $JSON::true,
```
Comment (kegsay) [inline lib/SyTest/Homeserver/Dendrite.pm:104]: This is in the wrong place. It's in global here but the dendrite pr adds this flag to the pushers config.

Comment (tommie) [reply lib/SyTest/Homeserver/Dendrite.pm]: Yepp, revised.

**Comment by tommie** [comment]
See https://github.com/matrix-org/dendrite/pull/1842#issuecomment-985752226 for the commit ordering relative Dendrite.

**Comment by tommie** [comment]
@PiotrKozimor raised the question if this can be merged before the config option has been added to Dendrite. I've assumed that Dendrite will fail if we try, but maybe that's not the case?

**Comment by tommie** [comment]
... and @neilalexander just answered it's okay on the Dendrite 1842.

**Comment by clokep** [comment]
I removed the review on synapse-core since I don't think it is necessary here. Please shout if you'd like us to take a look though!

**Review by neilalexander — APPROVED** [review]

File: lib/SyTest/Homeserver/Dendrite.pm [hunk 2]
```diff
@@ -195,37 +195,41 @@ sub _get_config
...
          database => {
-            connection_string => 
+            connection_string =>
                ( ! defined $ENV{'POSTGRES'} || $ENV{'POSTGRES'} == '0') ?
                "file:$self->{hs_dir}/mscs.db" : $db_uri,
          },
          mscs => ["msc2836", "msc2946", "msc2444", "msc2753"],
       },
 
+      push_server => {
```
Comment (PiotrKozimor) [inline lib/SyTest/Homeserver/Dendrite.pm:205]: @tommie I just realized that we already have `push_server` section below

Comment (neilalexander) [reply lib/SyTest/Homeserver/Dendrite.pm:205]: I've fixed that. Good catch.

Comment (tommie) [reply lib/SyTest/Homeserver/Dendrite.pm:205]: Sorry about that. Thank you!

**Pull request merged**